Versioner/AppStoreInfo.swift

```diff
@@ -0,0 +1,155 @@
+//
+//  AppStoreInfo.swift
+//  PrinceOfVersions
+//
+//  Created by Jasmin Abou Aldan on 06/02/2020.
+//  Copyright © 2020 Infinum Ltd. All rights reserved.
+//
+
+import Foundation
+
+#if canImport(UIKit)
+import UIKit
+#elseif canImport(AppKit)
+import AppKit
+#endif
+
+public struct AppStoreInfo{
+
+    // MARK: - Private configuration
+
+    /// Data parsed from the configuration file
+    private struct ConfigurationData {
+        var latestVersion: Version?
+        var minimumSdkForLatestVersion: Version?
+        var installedVersion: Version?
+        var sdkVersion: Version?
+    }
+
+    private var configurationData = ConfigurationData()
+    private var currentVersionReleaseDate: Date?
+
+
+    // MARK: - Init -
+
+    init(data: Data?, bundle: Bundle = Bundle.main) throws {
+        guard let data = data else {
+            throw PrinceOfVersionsError.invalidJsonData
+        }
+        let json = try JSONSerialization.jsonObject(with: data, options: []) as? NSDictionary
+
+        // JSON data
+        guard let value = json as? [String: AnyObject] else {
+            throw PrinceOfVersionsError.invalidJsonData
+        }
+
+        guard
+            let resultCount = value["resultCount"] as? Int,
+            resultCount > 0
+        else {
+            throw PrinceOfVersionsError.dataNotFound
+        }
+
+        guard
+            let arrayOfResults = value["results"] as? [[String: AnyObject]],
+            let results = arrayOfResults.first // Get only first result for queried bundle id
+        else {
+            throw PrinceOfVersionsError.invalidJsonData
+        }
+
+        // Latest version
+
+        if let versionString = results["version"] as? String {
+            configurationData.latestVersion = try Version(string: versionString)
+        } else {
+            throw PrinceOfVersionsError.invalidLatestVersion
+        }
+
+        if let minimumSdkForLatestVersionString = results["minimumOsVersion"] as? String {
+            configurationData.minimumSdkForLatestVersion = try? Version(string: minimumSdkForLatestVersionString)
+        }
+
+        // Installed version
+
+        guard let currentVersionString = bundle.object(forInfoDictionaryKey: "CFBundleShortVersionString") as? String else {
```

Comment:
I smell code duplications here. I would ge with extracting device data since it's the same for both methods.

Reply:
For `// Installed version` part - it is code duplication from `UpdateInfo`, but I have discussed with Android team that we will remove that from the PoV. 

If this will stay in the PoV in version 4.0, we will extract this into the separate struct

Reply:
😞